native/cocos/bindings/jswrapper/napi/Object.h

```diff
@@ -0,0 +1,352 @@
+#pragma once
+#include <cassert>
+#include "../RefCounter.h"
+#include "../Value.h"
+#include "../config.h"
+#include "CommonHeader.h"
+#include "HelperMacros.h"
+
+namespace se {
+class Class;
+class ObjectRef {
+private:
+    napi_ref   _ref       = nullptr;
+    int        _refCounts = 0;
+    napi_env   _env       = nullptr;
+    napi_value _obj       = nullptr;
+
+public:
+    ~ObjectRef() {
+        deleteRef();
+    }
+    napi_value getValue(napi_env env) const {
+        napi_value  result;
+        napi_status status;
+        NODE_API_CALL(status, env, napi_get_reference_value(env, _ref, &result));
+        assert(status == napi_ok);
+        assert(result != nullptr);
+        return result;
+    }
+    void initWeakref(napi_env env, napi_value obj) {
+        assert(_ref == nullptr);
+        _obj = obj;
+        _env = env;
+        napi_create_reference(env, obj, 0, &_ref);
+    }
+    void setWeakref(napi_env env, napi_ref ref) {
+        assert(_ref == nullptr);
+        _ref = ref;
+    }
+    void initStrongRef(napi_env env, napi_value obj) {
+        assert(_ref == nullptr);
+        _refCounts = 1;
+        _obj       = obj;
+        napi_create_reference(env, obj, _refCounts, &_ref);
+        _env = env;
+    }
+    void incRef(napi_env env) {
+        assert(_refCounts == 0);
+        if (_refCounts == 0) {
+            uint32_t result = 0;
+            _refCounts      = 1;
+            napi_reference_ref(env, _ref, &result);
```

Comment:
If incRef is invoked more than once, the _refCount is still 1, and defRef will always subtract 1. And defRef always does the napi_reference_unref thing. Consider the situlation:
```
obj.incRef();
obj.incRef();
obj.decRef();
obj.decRef()

```

Reply:
yes i will fixed, unref if ((_refCounts == 0))